tokio/src/runtime/park.rs

```diff
@@ -305,35 +305,55 @@ impl Inner {
         Arc::into_raw(this) as *const ()
     }
 
+    /// # Safety
+    ///
+    /// The pointer must have been created by [`Self::into_raw`].
     unsafe fn from_raw(ptr: *const ()) -> Arc<Inner> {
-        Arc::from_raw(ptr as *const Inner)
+        unsafe { Arc::from_raw(ptr as *const Inner) }
     }
 }
 
+// TODO: Is this really a unsafe function?
 unsafe fn unparker_to_raw_waker(unparker: Arc<Inner>) -> RawWaker {
     RawWaker::new(
         Inner::into_raw(unparker),
         &RawWakerVTable::new(clone, wake, wake_by_ref, drop_waker),
     )
 }
 
+/// # Safety
+///
+/// The pointer must have been created by [`Inner::into_raw`].
 unsafe fn clone(raw: *const ()) -> RawWaker {
-    Arc::increment_strong_count(raw as *const Inner);
-    unparker_to_raw_waker(Inner::from_raw(raw))
+    unsafe {
+        Arc::increment_strong_count(raw as *const Inner);
+    }
+    unsafe { unparker_to_raw_waker(Inner::from_raw(raw)) }
 }
 
+/// # Safety
+///
+/// The pointer must have been created by [`Inner::into_raw`].
 unsafe fn drop_waker(raw: *const ()) {
-    drop(Inner::from_raw(raw));
+    drop(unsafe { Inner::from_raw(raw) });
```

Comment:
This function is very short, and it’s clear enough for developers to reason about the safety requirements, so we don’t need additional "Safety" comments here”